assessment/libs/complex.php

```diff
@@ -0,0 +1,636 @@
+//          default is 3 decimal places. 
+// Returns:
+// A complex number in standard form as a string. If num has more than one complex number, the function returns
+// an array of the polar forms as strings: answer = array("r1 (cos(t1) + isin(t1))", "r2 (cos(t2) + isin(t2))", ...).
+
+function cx_format2pol(array $num, string $argin="rad", int $roundto=3) {
+
+    $A=array();
+    for ($i=0;$i<count($num);$i++){
+        $num1=array($num[$i]);
+        $sq=$num[0][0]**2+$num[0][1]**2;
+        $r= reduceradical($sq);
+        //$r= round(cx_modul($num1),$roundto);
+
+        if ($argin=="rad"){
+                $th0= cx_arg($num1);
+                $th1= round($th0/pi(),$roundto);  
+                $th1=makexxpretty("$th1 pi"); 
+        }
+            else {
+                $th1=round(cx_arg($num1,"deg"),$rounto);
+        }
+        
+        $A[$i]=makexpretty("$r (cos($th1)+isin($th1))");
+    }
+    
+    if (count($num)>1){
+        $ans=$A;
+    }    else{
+            $ans=$A[0];
+        }
+
+    return $ans;
+      
+}
+
+//------------------------------String Formatted Quadratic real and complex roots------------------------
+// Function: cx_prettyquadRoot(a,b,c, [roundto=3])
+// Returns an array of the string of roots of the quadratic equation f(x) = ax^2 + bx + c. Real roots are returned
+// as an array("r1","r2") ordered from the smallest to largest, and the complex roots are returned 
+// as an array("a+bi","a-bi"). This function is suitable for displaying answer (i.e., $showanswer).
+// 
+// Parameters:
+// a: The numerical coefficient of x^2
+// b: The numerical coefficient of x    
+// c: The constant
+// roundto: Optional - number of decimal places to which modulus should be rounded off; 
+//          default is 3 decimal places. 
+//
+// Returns:
+// An array of formatted string of roots (either real or complex) of the quadratic equation.
+
+function cx_prettyquadRoot(float $a, float $b, float $c, int $roundto = 3){
+    $d=$b**2 - 4*$a*$c;
+    
+    if ($d<0){
+        
+            $a2=$a*2;
+            $D=makereducedfraction($b,$a2);
+
+            $re= -$D;
+            $N=reduceradicalfrac(1,-$d,$a2);
+            $im= sqrt(abs($d))/(2*$a);
+            $im2=-$im;
+            $st=array("-$D + $N i","-$D - $N i");
+    }
+        else {
+            $r1=round(((-$b-sqrt($d))/(2*$a)),$roundto);
+            $r2=round(((-$b+sqrt($d))/(2*$a)),$roundto);
+            $st=array("$r1","$r2");
+        }
+
+    return $st;
+
+}
+
+//-------------------------------------Plot a complex number------------------------------------
+// Function: cx_plot(num, [argin = "deg" , roundto = 3, showlabels = True] )
+// Returns the plot of a complex number. It can be used to find z from plot as well as in solutions.
+// 
+// Parameters:
+// num: An array of real and imaginary parts of complex numbers given in square brackets: num = array([Re, Im]).
```

Comment:
Either change this to remove the double arrays, or modify the function to allow plotting multiple complex numbers.